bookwyrm/activitypub/base_activity.py

```diff
@@ -380,7 +380,23 @@ def resolve_remote_id(
         logger.info("Could not connect to host for remote_id: %s", remote_id)
         return None
     except requests.HTTPError as e:
-        logger.exception("HTTP error - remote_id: %s - error: %s", remote_id, e)
+        if (
+            hasattr(e, "response")
+            and hasattr(e.response, "status_code")
+            and e.response.status_code == 410
+        ):
+            # only log a warning for "gone" since there is not much we can do
+            logger.warning(
+                "request for object dropped because it is gone (410) - remote_id: %s",
+                remote_id,
+            )
+            # check whether the remote_id is a user and already in our DB
+            existing = models.User.find_existing_by_remote_id(remote_id)
+            # ensure deleted remote users are also deleted in our DB
+            if existing and not existing.deleted:
+                existing.delete()
```

Comment:
I also feel a bit nervous myself about this part, user deletion, which I don't fully understand:

- What is the intent here?
- Will previously existing content by this user (e.g., comments) still be displayed in our instance (and should it not happen)?

This action is never undone, I don't think; the 410 status code could be a remote-server hiccup, so that's makes me uneasy too.

Reply:
This is the bit that I'm a little uneasy about as well, though I do think it's justified. Basically what this is intended to do is delete the _local manifestation of a remote user_ in the local database if we know the remote user has been deleted.

In theory, this shouldn't be necessary because if a remote user is deleted we should receive an Activity. But given the original issue, it seems this isn't always the case, for whatever reason. I'm definitely open to other suggestions for handling this, the intent is mostly to avoid sending unnecessary requests to user inboxes we know don't exist.